src/pytest_servers/azure.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import time
+
+import pytest
+import requests
+
+AZURITE_PORT = 10000
+AZURITE_URL = "http://localhost:{port}"
+AZURITE_ACCOUNT_NAME = "devstoreaccount1"
+AZURITE_KEY = "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw=="  # noqa: E501
+AZURITE_ENDPOINT = f"{AZURITE_URL}/{AZURITE_ACCOUNT_NAME}"
+AZURITE_CONNECTION_STRING = f"DefaultEndpointsProtocol=http;AccountName={AZURITE_ACCOUNT_NAME};AccountKey={AZURITE_KEY};BlobEndpoint={AZURITE_ENDPOINT};"  # noqa: E501
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="session")
+def azurite(request, docker_client):
+    if docker_client is None:
+        logger.warning(
+            "Azurite cannot be started because docker is not available."
+        )
+        yield None
+        return
+
+    port = request.config.getoption("--azurite-port")
+
+    container = docker_client.containers.run(
+        "mcr.microsoft.com/azure-storage/azurite",
+        command="azurite-blob --loose --blobHost 0.0.0.0",
+        name="pytest-servers-azurite",
+        stdout=True,
+        stderr=True,
+        detach=True,
+        remove=True,
+        ports={port: 10000},
```

Comment:
We shouldn't hardcode the port mapping. See dvc-azure, where we automatically retrieve random port that was assigned for this docker container.